src/ray/core_worker/transport/actor_scheduling_queue.cc

```diff
@@ -107,22 +109,38 @@ void ActorSchedulingQueue::Add(
         task_spec,
         rpc::TaskStatus::PENDING_ACTOR_TASK_ARGS_FETCH,
         /* include_task_info */ false));
-    waiter_.Wait(dependencies, [seq_no, this]() {
-      RAY_CHECK(boost::this_thread::get_id() == main_thread_id_);
-      auto it = pending_actor_tasks_.find(seq_no);
-      if (it != pending_actor_tasks_.end()) {
-        const TaskSpecification &task_spec = it->second.TaskSpec();
-        RAY_UNUSED(task_event_buffer_.RecordTaskStatusEventIfNeeded(
-            task_spec.TaskId(),
-            task_spec.JobId(),
-            task_spec.AttemptNumber(),
-            task_spec,
-            rpc::TaskStatus::PENDING_ACTOR_TASK_ORDERING_OR_CONCURRENCY,
-            /* include_task_info */ false));
-        it->second.MarkDependenciesSatisfied();
-        ScheduleRequests();
+
+    auto in_actor_dependencies_metadata = task_spec.GetInActorDependenciesMetadata();
```

Comment:
`task_spec.GetInActorDependenciesMetadata();` is a subset of `task_spec.GetDependencies();`.